### PR TITLE
Added the ability to remove a block

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Added the ability to remove a block
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes PrestaShop/PrestaShop#15935
| How to test?  | * Add a new block item<br>* Check the front it is not here<br>* Enable it<br>* Check the front it is here<br>* Edit it<br>* Check the front it is here <br>* Remove it<br>* Check the front it is not here

### CODE REVIEW

**Comment by matks** [comment]
You removed 2700 lines 😄 is that intentional for this PR ?

**Comment by Progi1984** [comment]
> You removed 2700 lines smile is that intentional for this PR ?

Yes, Webpack compress to multiline into oneline now ;)

File: _dev/back/index.js [hunk 1]
```diff
@@ -121,6 +121,31 @@ $(window).ready(function () {
     }
   });
 
+  // Tab Content : Delete
+  $(document).on('click', '.psre-delete', function () {
+    var idBlock = $(this).data('id');
+    if (confirm(txtConfirmRemoveBlock)) {
```
Comment (PierreRambaud) [inline _dev/back/index.js]: You're also able to return `if (!confirm` to gain one indentation level :)

File: _dev/back/index.js [hunk 2]
```diff
@@ -121,6 +121,31 @@ $(window).ready(function () {
...
+      $.ajax({
+        type: 'POST',
+        dataType: 'JSON',
+        url: psr_controller_block_url,
+        data: {
+          ajax: true,
+          action: 'DeleteBlock',
+          idBlock: idBlock,
+        },
+        success: function (data) {
```
Comment (PierreRambaud) [inline _dev/back/index.js]: What about on failure?

File: views/templates/admin/tabs/content/listing.tpl [hunk 3]
```diff
@@ -41,8 +41,7 @@
                 </div>
                 <div class="listing-body col-lg-12  col-xs-12">
                     {foreach from=$allblock item=$block key=$key}
-                        <div data-id='' class="listing-general-rol row"
-                             data-block='{$block.id_psreassurance}'>
+                        <div class="listing-general-rol row" data-block='{$block.id_psreassurance}'>
```
Comment (PierreRambaud) [inline views/templates/admin/tabs/content/listing.tpl]:
```suggestion
                        <div class="listing-general-rol row" data-block="{$block.id_psreassurance}">
```

File: views/templates/admin/tabs/content/listing.tpl [hunk 4]
```diff
@@ -84,9 +83,13 @@
                                         <span class="switch_text switch-on" style="{if !$block['status']}display:none;{/if}">{l s='Activated' d='Modules.Blockreassurance.Admin'}</span>
                                         <span class="switch_text switch-off" style="{if $block['status']}display:none;{/if}">{l s='Deactivated' d='Modules.Blockreassurance.Admin'}</span>
                                     </label>
-
-                                    <span class="psre-edit" data-id="{$block['id_psreassurance']}"><i
-                                                class="material-icons">edit</i></span>
+                                  <div class="btn-group">
```
Comment (PierreRambaud) [inline views/templates/admin/tabs/content/listing.tpl]: look like there is something weird with the indentation level :thinking:

File: _dev/back/index.js (hunk 1 above)
Comment (Progi1984) [reply _dev/back/index.js]: :+1:

File: _dev/back/index.js (hunk 2 above)
Comment (Progi1984) [reply _dev/back/index.js]: :+1:

File: views/templates/admin/tabs/content/listing.tpl (hunk 4 above)
Comment (Progi1984) [reply views/templates/admin/tabs/content/listing.tpl]: :+1:

**Comment by matks** [comment]
Thank you @Progi1984

**Pull request merged**